src/service.py

```
@@ -461,6 +414,33 @@ def hw_tools() -> Set[HWTool]:
         return {HWTool.DCGM}
 
 
+class SmartCtlExporter(SnapExporter):
+    """A class representing the smartctl exporter and the metric endpoints."""
+
+    exporter_name: str = "smartctl-exporter"
+
+    def __init__(self, config: ConfigData) -> None:
+        """Initialize the SmartctlExporter class."""
+        self.port = int(config["smartctl-exporter-port"])
+        self.log_level = str(config["exporter-log-level"])
+        self.strategy = SmartCtlExporterStrategy(str(config["smartctl-exporter-snap-channel"]))
+        super().__init__(config)
+
+    @staticmethod
+    def hw_tools() -> Set[HWTool]:
+        """Return hardware tools to watch."""
+        return {HWTool.SMARTCTL_EXPORTER}
+
+    def configure(self) -> bool:
+        """Set the necessary exporter configurations or change snap channel."""
+        return super().configure() and self.set(
```

(out of scope for this PR) By the way, I can't see anything that actually uses the return value from configure, install, or uninstall.  That might be a nice refactoring soon - remove the return value.  I reckon we can rely on errors raised, rather than a return value.

---

the method name itself confuses me too: it's named `configure` but internally run `install`

---

That happens because the snap lib uses install for both install or refresh, so i would be ok changing the name to be more explicit about this

---

That could be a refactoring in the future